x-pack/plugins/snapshot_restore/server/test/helpers/router_mock.ts

```diff
@@ -0,0 +1,113 @@
+/*
```

Comment:
I wonder if we can get something like this available from core for all tests like this.

It would also be cool to test validation in something like this, there is a similar kind of hackery going on elsewhere (like in upgrade assistant), but I think this iteration is quite clean :+1:. Perhaps something we can open an issue for (if there is not one already).

Reply:
Let see first if more apps need this mock. I had to find a solution for SR as the app had so many written unit tests (and no API integration tests). 👍